frontend/src/pages/auth/login.tsx

```diff
@@ -214,20 +214,56 @@ export const LoginPage: FC = () => {
 
     const handleHostNameChange = useCallback((newHostName: string) => {
         if (databaseType.id !== DatabaseType.MongoDb || !newHostName.startsWith("mongodb+srv://")) {
-            return setHostName(newHostName);
-        }
-        const url = new URL(newHostName);
-        setHostName(url.hostname);
-        setUsername(url.username);
-        setPassword(url.password);
-        setDatabase(url.pathname.substring(1));
-        const advancedForm = { "Port": "27017", "URL Params": `?${url.searchParams.toString()}`, "DNS Enabled": "false" };
-        if (url.port.length === 0) {
-            advancedForm["Port"] = "";
-            advancedForm["DNS Enabled"] = "true";
+            // Checks the valid postgres URL
+            if (databaseType.id === DatabaseType.Postgres && (newHostName.startsWith("postgres://") || newHostName.startsWith("postgresql://"))) {
+                try {
+                    const url = new URL(newHostName); 
+                    const hostname = url.hostname;
+                    const username = url.username;
+                    const password = url.password;
+                    const database = url.pathname.substring(1);
+
+                    // gives warning
+                    if (!hostname || !username || !password || !database) {
+                        return notify("We could not extract all required details (host, username, password, or database) from this URL. Please enter the information manually.", "warning");
```

Comment:
Could you remove the "return" and just notify the user? That way, we do a best-effort processing and extract whatever we can.